src/entities/road.lua

```diff
@@ -86,17 +86,46 @@ function road:draw()
   love.graphics.setColor(255, 255, 255, 255)
 end
 
-function road:update(angle, roadShift)
+  
+function road:update(newSegments)
   -- create a road segment at the position and angle of the frontier.
-  self:addsegment()
+  for _, s in ipairs(newSegments) do
+  self:addSegment(s.x, s.y, s.angle)
+  end
+  local i = table.getn(newSegments)
```

Comment:
Minor nitpick, might make this a little more clear to do something like
```
local lastSegment = newSegments[#newSegments]
self:setFrontier(lastSegment.x, lastSegment.y, lastSegment.angle, "center")
```

Reply:
good point!